pkg/operator/controller/cabundle-configmap/controller.go

```diff
@@ -0,0 +1,166 @@
+	operatorCache := mgr.GetCache()
+	reconciler := &reconciler{
+		cache:  operatorCache,
+		client: mgr.GetClient(),
+		config: config,
+	}
+	c, err := controller.New(controllerName, mgr, controller.Options{
+		Reconciler: reconciler,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	mapFunc := func(ctx context.Context, o client.Object) []reconcile.Request {
+		return []reconcile.Request{
+			{
+				NamespacedName: config.OperatorNamespacedName,
+			},
+		}
+	}
+
+	matchesName := func(name types.NamespacedName) func(o client.Object) bool {
+		return func(o client.Object) bool {
+			return o.GetNamespace() == name.Namespace &&
+				o.GetName() == name.Name
+		}
+	}
+
+	// watch the Service CA configmap for any changes.
+	if err := c.Watch(
+		source.Kind(operatorCache, &corev1.ConfigMap{}),
+		handler.EnqueueRequestsFromMapFunc(mapFunc),
+		predicate.And(predicate.NewPredicateFuncs(matchesName(config.ServiceCAConfigMapName))),
+	); err != nil {
+		return nil, err
+	}
+
+	// watch the Admin CA configmap for any changes.
+	if err := c.Watch(
+		source.Kind(operatorCache, &corev1.ConfigMap{}),
+		handler.EnqueueRequestsFromMapFunc(mapFunc),
+		predicate.And(predicate.NewPredicateFuncs(matchesName(config.AdminCAConfigMapName))),
+	); err != nil {
+		return nil, err
+	}
+
+	// watch the Ingress CA configmap created by this controller for any changes.
+	if err := c.Watch(
+		source.Kind(operatorCache, &corev1.ConfigMap{}),
+		handler.EnqueueRequestsFromMapFunc(mapFunc),
+		predicate.And(predicate.NewPredicateFuncs(matchesName(config.IngressCAConfigMapName))),
+	); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+// Reconcile reconciles the `ingress-ca-bundle` configmap in the DefaultOperandNamespace namespace
+// which is created by combining the contents of `admin-ca-bundle` and `service-ca-bundle`
+// configmaps in GlobalUserSpecifiedConfigNamespace and DefaultOperandNamespace namespace.
+func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log.Info("reconciling ingress-ca-bundle", "request", request)
+
+	ic := &operatorv1.IngressController{}
+	if err := r.client.Get(ctx, request.NamespacedName, ic); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("ingresscontroller not found; reconciliation will be skipped", "request", request)
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get ingresscontroller %q: %w", request.NamespacedName, err)
+	}
+
+	if err := r.ensureIngressCABundleConfigMap(ctx, ic); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to ensure ingress CA bundle configmap: %w", err)
+	}
+
+	if ic.DeletionTimestamp == nil && !slice.ContainsString(ic.Finalizers, finalizer) {
+		// Ensure the ingresscontroller has a finalizer, so we get a
+		// chance to delete the configmap when the ingresscontroller is
+		// deleted.
+		return r.addFinalizer(ctx, finalizer, ic)
```

Comment:
I'm trying to understand the motivation of adding a finalizer to the default Ingress Controller.

Do we need a finalizer? I notice that the `service-ca-bundle` config map doesn't get a finalizer, and it seems similar to `ingress-ca-bundle`  because it's also a global configmap for all ingress controllers: https://github.com/openshift/cluster-ingress-operator/blob/ee583974c847c6f66ed25c7a9edb9ce0e304a11c/pkg/operator/controller/ingress/controller.go#L1027

It seems a bit funky that the global `ingress-ca-bundle` configmap (used by all ingress controllers) would get deleted if the default ingress controller gets deleted. The relationship created by the finalizers here seems questionable. 

I am not sure if this finalizer should be here, can you help me understand this design decision?